File: Selenium/Project/Activity9.java
package LMS_Project_Activities;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.testng.Assert;
import org.testng.Reporter;
import org.testng.annotations.AfterClass;
import org.testng.annotations.BeforeClass;
import org.testng.annotations.Test;

public class Activity9 {
    WebDriver driver;
    @BeforeClass
    public void setUp(){
        WebDriverManager.firefoxdriver().setup();
        driver = new FirefoxDriver();
        driver.get("https://alchemy.hguy.co/lms");
        Reporter.log("Homepage loaded");


    }

    @Test
    public void testMethod1(){
        Assert.assertEquals(driver.getTitle(), "Alchemy LMS – An LMS Application");

        driver.findElement(By.xpath("//a[text()='All Courses']")).click();

        Assert.assertEquals(driver.getTitle(), "All Courses – Alchemy LMS");
        Reporter.log("Newpage loaded");
    }

    @Test
    public void testMethod2() {
        driver.findElement(By.xpath("//p//a[@href='https://alchemy.hguy.co/lms/courses/social-media-marketing/']")).click();
        Assert.assertEquals(driver.getTitle(), "Social Media Marketing – Alchemy LMS");
        driver.findElement(By.xpath("//a[text()='Login to Enroll']")).click();

        WebElement userName = driver.findElement(By.id("user_login"));
        WebElement passWord = driver.findElement(By.id("user_pass"));
        WebElement loginButton = driver.findElement(By.id("wp-submit"));

        userName.clear();
        userName.sendKeys("root");
        passWord.clear();
        passWord.sendKeys("pa$$w0rd");
        loginButton.click();
    }

    @Test
    public void testMethod3(){
        WebElement contentLink = driver.findElement(By.xpath("//div//a[@href='https://alchemy.hguy.co/lms/lessons/developing-strategy/']"));
        contentLink.click();
        Assert.assertEquals(driver.getTitle(), "Developing Strategy – Alchemy LMS");

        String completionStatus  = driver.findElement(By.xpath("//span[@class='ld-lesson-list-progress']")).getText();
        Assert.assertEquals(completionStatus, "100% COMPLETE");


        //logout
        driver.findElement(By.xpath("//span//img[contains(@class,'avatar-96')]")).click();
        driver.findElement(By.xpath("//a[text()='Logout']")).click();

        Reporter.log("Newpage loaded");
    }


    @AfterClass
    public void tearDown(){
        Reporter.log("browser closed");
        driver.quit();
    }
}

File: Selenium/Project/Activity9.py
from selenium.webdriver.common.by import By
# Set up the Firefox Driver with WebDriverManager
from selenium.webdriver.firefox.webdriver import WebDriver
from selenium.webdriver.firefox.service import Service as FirefoxService
from webdriver_manager.firefox import GeckoDriverManager

service = FirefoxService(GeckoDriverManager().install())

# Start the Driver
with WebDriver(service=service) as driver:
    # Navigate to the URL
    driver.get("https://alchemy.hguy.co/lms")

    # Print the title of the page
    title = driver.title
    print("Page title is:", title)
    assert title == "Alchemy LMS – An LMS Application"
    # Course page
    driver.find_element(By.XPATH, "//p//a[@href='https://alchemy.hguy.co/lms/courses/social-media-marketing/']").click()
    coursePage = driver.title
    print("Contact Title is ", coursePage)
    assert coursePage == "Social Media Marketing – Alchemy LMS"

    #login

    driver.find_element(By.XPATH, "//a[text()='Login to Enroll']").click()
    driver.find_element(By.ID, "user_login").send_keys("root")
    driver.find_element(By.ID, "user_pass").send_keys("pa$$w0rd")
    driver.find_element(By.ID, "wp-submit").click()

    #content
    contentLink = driver.find_element(By.XPATH, "//div//a[@href='https://alchemy.hguy.co/lms/lessons/developing-strategy/']")
    contentLink.click()
    contentPageTitle = driver.title
    assert contentPageTitle == "Developing Strategy – Alchemy LMS"

    #status
    completionStatus = driver.find_element(By.XPATH, "//span[@class='ld-lesson-list-progress']")
    print("Completion Status: ", completionStatus.text)
    assert completionStatus.text == "100% COMPLETE"

    #logout
    driver.find_element(By.XPATH, "//span//img[contains(@class,'avatar-96')]").click()
    driver.find_element(By.XPATH, "//a[text()='Logout']").click()
